modin/utils.py

```diff
@@ -40,6 +40,34 @@
 PANDAS_API_URL_TEMPLATE = f"https://pandas.pydata.org/pandas-docs/version/{pandas.__version__}/reference/api/{{}}.html"
 
 
+# Extension of NamedTuple has to be a little more complicated in order to avoid accidentally sharing
+# the `kwargs` dict across different instances if it is not passed to the constructor
+class Invocable(
+    NamedTuple("Invocable", [("func", Callable), ("args", Tuple), ("kwargs", dict)])
+):
+    """
+    Class for config data manipulating of exact type.
+
+    Parameters
+    ----------
+    func : Callable or future-like
+        The function to invoke. This is either a Python `Callable`, or a future/promise from a the computation engine.
+    args : Tuple, optional
+        Tuple of positional arguments to call func with.
+    kwargs : dict, optional
+        Dictionary of keyword arguments to call func with.
+    """
+
+    __slots__ = ()
+
+    def __new__(cls, func, args=None, kwargs=None):  # noqa: GL08
+        if args is None:
+            args = ()
+        if kwargs is None:
+            kwargs = {}
+        return super().__new__(cls, func, args, kwargs)
```

Comment:
```suggestion
        return super().__new__(cls, func, args or (), kwargs or {})
```

Reply:
Wouldn't this technically change the behavior if the caller of `Invocable` modifies args/kwargs after the fact? For example,
```
func = ... # something random
kwargs = {}
obj = Invocable(func=func, kwargs=kwargs)
kwargs["key1"] = 0
```
In your suggested change, since `{}` is falsey, the `Invocable` would get a new dict as its field, whereas in the current code it would share a reference to the originally passed-in dictionary.
This case is pretty pathological and I don't think we'd have situations where args/kwargs get modified after this constructor is called, but I'd just like to check whether it's reasonable to program defensively here.